run.go

```diff
@@ -638,6 +607,58 @@ func (b *Builder) addNetworkConfig(rdir, hostPath string, chownOpts *idtools.IDP
 	return cfile, nil
 }
 
+// generateHosts creates a containers hosts file
+func (b *Builder) generateHosts(rdir, hostname string, addHosts []string, chownOpts *idtools.IDPair) (string, error) {
+	hostPath := "/etc/hosts"
+	stat, err := os.Stat(hostPath)
+	if err != nil {
+		return "", errors.Wrapf(err, "error statting %q for container %q", hostPath, b.ContainerID)
+	}
+
+	orig, err := ioutil.ReadFile(hostPath)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to read %s", hostPath)
+	}
+	hosts := "# Generated by Buildah\n" + string(orig)
+	for _, host := range addHosts {
+		// verify the host format
+		values := strings.SplitN(host, ":", 2)
+		if len(values) != 2 {
+			return "", errors.Errorf("unable to parse host entry %q: incorrect format", host)
+		}
+		if values[0] == "" {
+			return "", errors.Errorf("hostname in host entry %q is empty", host)
+		}
+		if values[1] == "" {
+			return "", errors.Errorf("IP address in host entry %q is empty", host)
+		}
+		hosts += fmt.Sprintf("%s\t%s\n", values[1], values[0])
+	}
+
+	if hostname != "" {
+		hosts += fmt.Sprintf("127.0.0.1   %s\n", hostname)
+		hosts += fmt.Sprintf("::1         %s\n", hostname)
+	}
+	cfile := filepath.Join(rdir, filepath.Base(hostPath))
+	if err = ioutils.AtomicWriteFile(cfile, []byte(hosts), stat.Mode().Perm()); err != nil {
```

Comment:
I think `bytes.Buffer` is the right type for `hosts`.

Reply:
@rhatdan thoughts on @vrothberg 's comment here?

Reply:
@TomSweeneyRedHat, we can defer that to another PR. At some point, I would love to analyze the code for such and similar issues in the hope to further squeeze some cycles :)

Reply:
Moved to bytes.Buffer,  Is there something better then using the fmt.Sprintf?